keras/src/backend/torch/trainer.py

```diff
@@ -420,6 +420,20 @@ def append_to_outputs(batch_outputs, outputs):
                 )
             return outputs
 
+        if self.jit_compile:
+            # torch.compile() requires torch_param being registered. Thus,
+            # as symbolic build is needed to make sure all layers variables are
+            # initialized before invoke torch.compile(). _symbolic_build()
+            # should refactored to not require _compile_metrics and optimizer
+            # is defined.
```

Comment:
> _symbolic_build() should refactored to not require _compile_metrics and optimizer is defined.

Can you explain?

Reply:
`_symbolic_build` requires https://github.com/keras-team/keras/blob/master/keras/src/trainers/trainer.py#L1005-L1014 and these attributes are added during `model.compile`.

in this change, for torch params to work properly the requirements is that all layers are built before making any torch specific api calls. the easiest way to achieve this inside the trainer is to ensure `_symbolic_build` is called for all cases that `jit_compile` is true since `torch.compile` will call `named_parameters()` to compile the graph.

during unit test, i have found few unit tests breaks due to not calling `torch.compile` but requires calling `named_parameters` and it fails during `_symbolic_build` due to missing `compile_metrics`, `compile_loss` and `optimizer` attribute. i can try remove these and fixing the unit test instead.